packages/contracts-bedrock/test/L2/L1Block.t.sol

```diff
@@ -148,3 +191,197 @@ contract L1BlockEcotone_Test is L1BlockTest {
         assertEq(data, expReturn);
     }
 }
+
+contract L1BlockInterop_Test is L1BlockTest {
+    /// @dev Tests that setL1BlockValuesInterop updates the values appropriately.
+    function testFuzz_setL1BlockValuesInterop_succeeds(
+        uint32 baseFeeScalar,
+        uint32 blobBaseFeeScalar,
+        uint64 sequenceNumber,
+        uint64 timestamp,
+        uint64 number,
+        uint256 baseFee,
+        uint256 blobBaseFee,
+        bytes32 hash,
+        bytes32 batcherHash,
+        uint256[] calldata dependencySet
+    )
+        external
+    {
+        vm.assume(dependencySet.length == uint256(uint8(dependencySet.length)));
+
+        bytes memory functionCallDataPacked = Encoding.encodeSetL1BlockValuesEcotone(
+            baseFeeScalar, blobBaseFeeScalar, sequenceNumber, timestamp, number, baseFee, blobBaseFee, hash, batcherHash
+        );
+
+        vm.prank(depositor);
+        (bool success,) = address(l1Block).call(functionCallDataPacked);
+        assertTrue(success, "Function call failed");
+
+        functionCallDataPacked = Encoding.encodeSetL1BlockValuesInterop(
```

Comment:
nit: this would be easier to read/assure that it is correct if it used python kwargs style syntax, ie

```solidity
Encoding.encodeSetL1BlockValuesInterop({
  _baseFeeScalar: baseFeeScalar,
  ...
})
```

This guarantees args are passed in the correct order